Automattic/jetpack
# My Jetpack: allow plugins to override a product class

#### Changes proposed in this Pull Request:

* Adds a filter that allows plugins to override the class used by My Jetpack to represent a product

This will allow Search (and other plugins) to take over how their product behave in My Jetpack once the plugin is active.

#### Jetpack product discussion


p9dueE-55u-p2

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Add this to your theme functions.php or to a plugin:
```PHP
class Mock_Boost_Child extends Automattic\Jetpack\My_Jetpack\Products\Boost {
	public static function get_name() {
		return 'Child Boost';
	}
}

add_filter( 'my_jetpack_products_classes', 'my_filter' );

function my_filter( $classes ) {
  $classes['boost'] = 'Mock_Boost_Child';
  return $classes;
}
```
* Visit My Jetpack and make sure the Boost card now says "Child Boost"
* Make sure all tests pass
